lib/metasploit/cache/module/path/association_extension.rb

```diff
@@ -0,0 +1,71 @@
+# Adds {#each_changed} method to {Metasploit::Cache::Module::Path} associations, so that
+# {Metasploit::Cache::Module::Ancestor} from the given association can be populated from the file system.
+module Metasploit::Cache::Module::Path::AssociationExtension
+  # @note The yielded {Metasploit::Cache::Module::Ancestor} may contain unsaved changes.  It is the responsibility
```

Comment:
I've have to try some `@param <name> (see foo)` styles to see if I can copy `@overloads` correctly.

It may also be the case that when we integrate with metasploit-framework `Ancestor.each_changed` is not used and can be removed because the class and instance-level API is segregated by module type and payload type and there can't be one API that enumerates all of them since there isn't STI at the instance level and the associations have different names at the class level.